lib/lambda_ethereum_consensus/engine/execution.ex

```diff
@@ -0,0 +1,35 @@
+defmodule LambdaEthereumConsensus.Engine.Execution do
+  alias LambdaEthereumConsensus.RPC
+
+  # Default Execution Layer endpoint
+  @execution_engine_endpoint "http://0.0.0.0:8551"
+
+  # Default Execution Layer RPC version
+  @execution_engine_rpc_version "2.0"
+
+  @spec engine_exchange_capabilities(map) :: {:error, any} | {:ok, any}
+  def engine_exchange_capabilities(params \\ %{"capabilities" => []}) do
```

Comment:
I wouldn't add a default params, instead we should validate that the params have the format that we expect.

Reply:
Is it by adding guard clauses OR function definition params OR other ?

Guard clauses like this: 
```
def call(method, endpoint, version, params) when is_binary(method) do
end
```

Function definitions like this: 
```
def call({:ok, data}, data) do
end
```

These are examples

Reply:
You can deconstruct them like
`{"capabilities" => capabilities } = params` so you ensure that the params map has the fields that you are interesed in and you also don't do a passthrough

Reply:
We may just go

```elixir
@typespec engine_exchange_capabilities(list) :: {:ok, any} | {:error, any} 
def engine_exchange_capabilities(capabilities \\ []) do
```

ONLY IF an empty array is a valid param. If not:

```elixir
@typespec engine_exchange_capabilities(list) :: {:ok, any} | {:error, any} 
def engine_exchange_capabilities(capabilities) do
```

And then we can just build the map in the call. In general, when building an API wrapper it's a good idea to pass actual parameters instead of a map. Specially because the idea is to make this functions easier to understand and call from other parts of the project and arguments make that clear.

Reply:
In this instance an empty map would be a valid param for `engine_exchangeCapabilities` but wouldn't be right for other methods.

Do you recommend to not use default params at all then ?